components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -190,9 +195,9 @@ internal class AutoPushObserver(
         // Ignore push messages that do not have data.
         val rawEvent = message ?: return
 
-        accountManager.withConstellation {
-            processRawEventAsync(String(rawEvent))
-        }
+        accountManager.withConstellation { CoroutineScope(Dispatchers.Main).launch {
+            processRawEvent(String(rawEvent))
```

Comment:
Looks like we maybe want to make these observer methods `suspend` functions as well, so we don't need to context switch?

Reply:
Yeah, this will run as before in practice (underneath we have a dedicated dispatcher), and marking the interface funs as `suspend` will avoid switching and certainly make the intention here clearer.

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/7899

Reply:
@grigoryk I don't follow why we need to switch to the main dispatcher here. Previously the FxaPushSupportFeature would execute the call on any dispatcher since account manager had it's own dispatcher. Shouldn't that still be the case?